src/core/device.c

```diff
@@ -332,6 +332,15 @@ static void device_catchup(Unit *u) {
         Device *d = ASSERT_PTR(DEVICE(u));
 
         /* Second, let's update the state with the enumerated state */
+
+        /* If Device.found (set by Device.deserialized_found) does not have DEVICE_FOUND_UDEV, and the device
+         * has never been processed by udevd, then also drop the flag from Device.enumerated_found. Note, do
+         * not unset the flag when the device was previously processed by udevd. Otherwise, e.g. if
+         * systemd-udev-trigger.service is started just before reexec, reload, and so on, many devices have
+         * ID_PROCESSING=1 property on enumeration and will enters dead state. See issue #35329. */
```

Comment:
Hmm, this can use better wording, maybe:

"'"
If Device.found (set from Device.deserialized_found) does not have DEVICE_FOUND_UDEV, and the device has not been processed by udevd while enumeration, it indicates the unit was never active before reexecution, hence we can safely drop the flag from Device.enumerated_found. The device will be set up later when udev finishes processing (see also comment in device_setup_devlink_unit_one()).

Note 💣, if Device.found already contains udev, i.e. the unit was fully ready before reexecution, do not unset the flag. Otherwise, e.g. if systemd-udev-trigger.service is started just before reexec, reload, and so on, devices being reprocessed (carrying ID_PROCESSING=1 property) on enumeration and will enter dead state. See issue #35329.
"""